packages/core/chrome/core-chrome-browser-internal/src/project_navigation/project_navigation_service.ts

```diff
@@ -88,4 +106,52 @@ export class ProjectNavigationService {
       },
     };
   }
+
+  private setActiveProjectNavigationNodes(_location?: Location) {
+    if (!this.application) return;
+    if (!Object.keys(this.projectNavigationNavTreeFlattened).length) return;
+
+    const location = _location ?? this.application.history.location;
+    let currentPathname = this.http?.basePath.prepend(location.pathname) ?? location.pathname;
+
+    // We add possible hash to the current pathname
+    // e.g. /app/kibana#/management
+    currentPathname = stripQueryParams(`${currentPathname}${location.hash}`);
+
+    const activeNodes = findActiveNodes(
+      currentPathname,
+      this.projectNavigationNavTreeFlattened,
+      location
+    );
+
+    // Each time we call findActiveNodes() we create a new array of activeNodes. As this array is used
+    // in React in useCallback() and useMemo() dependencies arrays it triggers an infinite navigation
+    // tree registration loop. To avoid that we only notify the listeners when the activeNodes array
+    // has actually changed.
+    let requiresUpdate = activeNodes.length !== this.activeNodes$.value.length;
+
+    if (!requiresUpdate) {
+      for (const [i, nodesBranch] of this.activeNodes$.value.entries()) {
+        for (const [j, node] of nodesBranch.entries()) {
+          if (!deepEqual(node, activeNodes[i][j])) {
```

Comment:
why not compare the whole arrays? this would also cover case in case order of branches has changed for some reason 


const requiresUpdate = deepEqual(activeNodes, this.activeNodes$.value)


I think that function is smart enough to early exit in case the array length doesn't match

Reply:
Good point. Fixed in https://github.com/elastic/kibana/pull/159906/commits/252f1f62433dfea70d4507345cbeb310628d5a59